### PR TITLE
Help Center: Show support chat if user is admin and non owner on atomic site

### PR DESCRIPTION
**Pull request by agrullon95**
Related to https://github.com/Automattic/wp-calypso/issues/89984

## Proposed Changes

* These changes check if the user is on an atomic site and the user is an administrator on the site to grant them support within the Help Center.

## Why are these changes being made?

* Administrators on Atomic sites who are not site owners are currently not able to access the Zendesk chat. Instead, they are redirected to the forum form.

## Testing Instructions

* Checkout branch
* Sandbox public api and test site.
* Navigate to `apps/editing-toolkit`, build the code, and compress the `editing-toolkit-plugin` folder.
* Sync the code to your sandbox.
* Add a new user to the site with an administrator role.
* Open the Help Center and verify the `Still need help` button navigates to the correct support screen.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/help-center/src/hooks/use-user-is-admin-atomic.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+import { HELP_CENTER_STORE } from '../stores';
+
+export default function useUserIsAtomicAdmin() {
+	const currentSite = useSelect( ( select ) => {
+		const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
+		return helpCenterSelect.getSite();
+	}, [] );
+	const isAtomicSite = currentSite?.is_wpcom_atomic;
+	const currentUser = useSelector( getCurrentUser );
```
Comment (heavyweight) [inline packages/help-center/src/hooks/use-user-is-admin-atomic.ts:16]: In the context of the admin bar, for an atomic site we don't have this info.
The currentUser will be empty

**Comment by heavyweight** [comment]
Checkout how we set the site in the store here:
https://github.com/Automattic/wp-calypso/blob/afa39382660982914f7f356c4de8ded5c101b169/packages/help-center/src/components/help-center.tsx#L105
This way the data is available in the admin bar as well.
We would need to do something similar here

**Comment by kriskarkoski** [comment]
To clarify, admins on a site on any plan should be eligible for support. I'm not sure if that is working on non-Atomic plans (I'm going to do some testing), but wanted to note it. It's fine to ship this as it's the most urgent issue since Atomic sites can be down and are higher priority, but if it changes how we'd implement it to look at all plans, not just Atomic, that's fine.

**Comment by alshakero** [comment]
Note that I just merged an alternative fix to this issue in https://github.com/Automattic/wp-calypso/pull/91529. I made the endpoint make admins on any Atomic sites eligible for support.

**Comment by escapemanuele** [comment]
I believe this can be closed now @agrullon95, thanks for an alternate working solution 👍

**Pull request closed without merging**